### DISCUSSION TITLE
Welcome to chainfusion-starter Discussions!

### DISCUSSION THREAD

**Discussion by letmejustputthishere** [letmejustputthishere/icp-evm-coprocessor-starter#10] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by marceljuenemann** [comment]
@letmejustputthishere thanks for putting this project together, Moritz! I really think most of the things done off-chain these days could be moved on-chain using this tech. So I'm surprised to be the first one commenting here even :)

I've been working on  [:snowflake: Frosty Functions](https://github.com/marceljuenemann/frosty-functions) the last month, which aims to take this one step further by providing a Coprocessor-as-a-service. Instead of having to learn Rust and deploy your own canister, EVM developers can just write a Frosty Function in AssemblyScript, which is then interpreted sandboxed with wasmi in the Frosty canister (which will be verifiable and non-upgradable in the future). That way you also don't need to deploy a new bridge contract or buy ICP or cycles. Developers still get full control over an EVM wallet as I'm using the derivation path feature to generate wallets based on the function and caller. I'm also designing this to be horizontally scalable in the future, as outlined in my [building planet-scale apps on ICP](https://forum.dfinity.org/t/building-planet-scale-apps-on-icp/59346) post. 

I recorded an [early demo](https://drive.google.com/file/d/1pnjRNsjfdiGSVFzueKU_nHnjLtfAYkGf/view) and hope to have an MVP ready soon, but would very much appreciate any feedback on the idea!
